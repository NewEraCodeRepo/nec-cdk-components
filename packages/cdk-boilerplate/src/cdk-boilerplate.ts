#!/usr/bin/env node
import * as cloudtrail from '@aws-cdk/aws-cloudtrail';
import { Bucket } from '@aws-cdk/aws-s3';
import { Construct } from '@aws-cdk/core';

export interface ICDKBoilerplateProps {
  bucketName: string;
}

/**
 * Boilerplate
 */
export class CDKBoilerplate extends Construct {
  public bucketWebsiteUrl: string;

  constructor(scope: Construct, id: string, props: ICDKBoilerplateProps) {
    super(scope, id);

    const bucket = new Bucket(this, 'CloudTrailBucket', {
      bucketName: props.bucketName,
    });
    // tslint:disable-next-line:no-unused-expression
    new cloudtrail.Trail(this, 'CloudTrail', {
      bucket,
      isMultiRegionTrail: true,
    });

    this.bucketWebsiteUrl = bucket.bucketWebsiteUrl;
  }
}
