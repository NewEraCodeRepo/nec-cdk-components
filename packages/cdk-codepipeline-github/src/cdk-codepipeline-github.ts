import { IStage, Pipeline } from '@aws-cdk/aws-codepipeline';
import { LambdaFunction } from '@aws-cdk/aws-events-targets';
import { PolicyStatement } from '@aws-cdk/aws-iam';
import { Code, Function, Runtime } from '@aws-cdk/aws-lambda';
import { Construct, Duration, SecretValue } from '@aws-cdk/core';
import { join } from 'path';

export interface ICdkCodepipelineGithubProps {
  /**
   * The Codepipeline stage publishing status events
   */
  readonly stage?: IStage;

  /**
   * The Codepipeline publishing status events
   */
  readonly pipeline?: Pipeline;

  /**
   * The arn of a secret in the Secrets Manager with
   * a github token value.
   */
  readonly gitHubSecretArn: string;
}

/**
 * Publishes status events from a stage in Codepipeline
 * to a Github repo using webhooks.
 *
 * SAM template conversion by Julia Jacobs
 * {@see {@link https://github.com/jewelsjacobs}}
 *
 * SAM template created by Jens Eickmeyer
 * {@see {@link https://github.com/jenseickmeyer/github-commit-status-bot}}
 * {@see {@link https://scratchpad.blog/devops/howto/send-commit-status-from-codepipeline-to-github/}}
 *
 * @example <caption>Example usage of construct in pipeline.</caption>
 * // One possible way to get an existing secret's arn - using AwsCustomResource
 * // and AWS SecretsManager SDK
 * const getGithubSecret = new AwsCustomResource(this, 'GetGithubSecret', {
 *   onUpdate: {
 *      service: 'SecretsManager',
 *      action: 'describeSecret',
 *      parameters: {
 *         SecretId: 'TOKENS/PIPELINE/GithubToken' // ID of secret
 *      }
 *   }
 * });
 * // ARN is a property in the describeSecret SDK response
 * const gitHubSecretArn = `${getGithubSecret.getData("ARN")}` as string;
 *
 * // pipeline
 * const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
 *   restartExecutionOnUpdate: true
 * });
 *
 * // source stage
 * pipeline.addStage({
 *   stageName: 'Source',
 *   actions: [sourceAction]
 * });
 *
 * // build stage
 * const buildStage = pipeline.addStage({
 *   stageName: 'Build',
 *   actions: [buildAction]
 * });
 *
 * // Github status checks based on status of the build stage
 * new CdkCodepipelineGithub(this, 'GithubCodepipeline', {
 *   stage: buildStage,
 *   gitHubSecretArn
 * });
 *
 * // deploy stage
 * pipeline.addStage({
 *   stageName: 'Deploy',
 *   actions: [deployAction]
 * });
 *
 * // or check status of pipeline itself:
 * new CdkCodepipelineGithub(this, 'GithubCodepipeline', {
 *   pipeline,
 *   gitHubSecretArn
 * });
 */
export class CdkCodepipelineGithub extends Construct {
  constructor(scope: Construct, id: string, props: ICdkCodepipelineGithubProps) {
    super(scope, id);

    const { stage, pipeline, gitHubSecretArn } = props;

    const statusLambda = new Function(this, 'StatusLambda', {
      code: Code.asset(join(__dirname, '..', 'lambda_packages', 'package.zip')),
      environment: {
        ACCESS_TOKEN: SecretValue.secretsManager(gitHubSecretArn).toString(), // github token passed to Lambda
      },
      handler: 'index.handler',
      runtime: Runtime.NODEJS_10_X,
      timeout: Duration.seconds(300),
    });

    const lambdaTarget = new LambdaFunction(statusLambda);

    if (stage) {
      stage.onStateChange('StageOnStateChange', lambdaTarget, {
        eventPattern: {
          detail: {
            state: ['STARTED', 'SUCCEEDED', 'FAILED'],
          },
        },
      });
    }

    if (pipeline) {
      pipeline.onStateChange('StageOnStateChange', {
        eventPattern: {
          detail: {
            state: ['STARTED', 'SUCCEEDED', 'FAILED'],
          },
        },
        target: lambdaTarget,
      });
    }

    if (!stage && !pipeline) {
      throw Error('Choose a pipeline or stage to send event status messages from');
    }

    statusLambda.addToRolePolicy(
      new PolicyStatement({
        actions: ['*'],
        resources: ['*'],
      }),
    );
  }
}
