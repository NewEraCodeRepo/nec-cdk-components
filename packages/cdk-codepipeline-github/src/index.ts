export * from './cdk-codepipeline-github';
