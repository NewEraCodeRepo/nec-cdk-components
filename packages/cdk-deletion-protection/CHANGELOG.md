# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.9.1](https://github.com/NewEraCodeRepo/nec-cdk-components/compare/v1.9.0...v1.9.1) (2020-03-02)

**Note:** Version bump only for package @neweracode/cdk-deletion-protection





# 1.9.0 (2020-03-02)


### Features

* updates scripts and docs with awsume and npm org ([aa87ab1](https://github.com/NewEraCodeRepo/nec-cdk-components/commit/aa87ab1d6da201479c4161efdc6ac3f98115126d))
* updates tests and codebuild ([ec42954](https://github.com/NewEraCodeRepo/nec-cdk-components/commit/ec429540c32942c6546b4c7b489533ee48973835))
