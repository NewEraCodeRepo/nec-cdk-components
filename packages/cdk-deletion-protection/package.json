{
  "name": "@neweracode/cdk-deletion-protection",
  "version": "1.8.3",
  "description": "An AWS CDK construct that protects infrastructure from deletion",
  "author": "Julia Jacobs <jjacobs2000@me.com>",
  "homepage": "https://github.com/NewEraCodeRepo/nec-cdk-components#readme",
  "license": "MIT",
  "main": "lib/cdk-deletion-protection.js",
  "publishConfig": {
    "access": "restricted"
  },
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/NewEraCodeRepo/nec-cdk-components.git"
  },
  "scripts": {
    "tsc": "tsc",
    "watch": "tsc -w",
    "cdk": "cdk",
    "test": "echo \"Error: run tests from root\" && exit 1"
  },
  "bugs": {
    "url": "https://github.com/NewEraCodeRepo/nec-cdk-components/issues"
  },
  "devDependencies": {
    "@aws-cdk/assert": "^1.20.0",
    "@babel/cli": "^7.7.7",
    "@babel/core": "^7.7.7",
    "@babel/preset-env": "^7.7.7",
    "@babel/preset-typescript": "^7.7.7",
    "@types/jest": "^24.0.25",
    "@types/node": "^13.1.6",
    "aws-cdk": "^1.20.0",
    "babel-core": "7.0.0-bridge.0",
    "jest-cdk-snapshot": "^1.2.0",
    "mock-fs": "^4.10.4",
    "ts-jest": "^24.3.0",
    "ts-node": "^8.6.1",
    "tslint": "^5.20.1",
    "tslint-config-prettier": "^1.18.0",
    "typescript": "^3.7.4"
  },
  "dependencies": {
    "@aws-cdk/core": "^1.20.0",
    "@aws-cdk/custom-resources": "^1.20.0",
    "source-map-support": "^0.5.16"
  }
}
