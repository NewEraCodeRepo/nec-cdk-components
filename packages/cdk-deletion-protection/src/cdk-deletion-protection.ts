#!/usr/bin/env node
import { CfnDeletionPolicy, CfnResource, Construct } from '@aws-cdk/core';
import { AwsCustomResource } from '@aws-cdk/custom-resources';

export interface ICDKDeletionProtectionProps {
  resourceDeletionPolicy?: string; // "DELETE" | "RETAIN" | "SNAPSHOT"
  terminationProtection?: boolean;
}

/**
 * DeletionProtection
 *
 * Example usage:
 *
 * const deletionProtection = new CDKDeletionProtection(this, 'ProtectionPolicies', {
 *    resourceDeletionPolicy: "RETAIN",
 *    terminationProtection: true,
 * });
 *
 * // enable stack termination deletion protection:
 * deletionProtection.setStackTerminationProtection(this.stackName);
 *
 * // protect a Cognito User Pool from getting deleted on update:
 * const userPool = new UserPool(this, 'cdkUserPool', {...});
 * const cfnUserPool = userPool.node.defaultChild as CfnUserPool;
 * deletionProtection.setResourcePolicy(cfnUserPool);
 */
export class CDKDeletionProtection extends Construct {
  public setResourcePolicy: (cfnResource: CfnResource) => void;
  public setStackTerminationProtection: (stackName: string) => void;
  constructor(scope: Construct, id: string, props?: ICDKDeletionProtectionProps) {
    super(scope, id);

    this.setResourcePolicy = (cfnResource: CfnResource) => {
      if (props && props.hasOwnProperty('resourceDeletionPolicy')) {
        switch (props.resourceDeletionPolicy) {
          case 'DELETE':
            cfnResource.cfnOptions.updateReplacePolicy = CfnDeletionPolicy.DELETE;
            cfnResource.cfnOptions.deletionPolicy = CfnDeletionPolicy.DELETE;
            break;
          case 'RETAIN':
            cfnResource.cfnOptions.updateReplacePolicy = CfnDeletionPolicy.RETAIN;
            cfnResource.cfnOptions.deletionPolicy = CfnDeletionPolicy.RETAIN;
            break;
          case 'SNAPSHOT':
            cfnResource.cfnOptions.updateReplacePolicy = CfnDeletionPolicy.SNAPSHOT;
            cfnResource.cfnOptions.deletionPolicy = CfnDeletionPolicy.SNAPSHOT;
            break;
          default:
            cfnResource.cfnOptions.updateReplacePolicy = CfnDeletionPolicy.RETAIN;
            cfnResource.cfnOptions.deletionPolicy = CfnDeletionPolicy.RETAIN;
            break;
        }
      } else {
        throw new Error('Resource deletion policy required.');
      }
    };

    this.setStackTerminationProtection = (stackName: string) => {
      if (props && props.hasOwnProperty('terminationProtection')) {
        // Enable stack termination protection
        // using the AWS SDK CloudFormation updateTerminationProtection API:
        // https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CloudFormation.html#updateTerminationProtection-property

        // tslint:disable-next-line:no-unused-expression
        new AwsCustomResource(this, 'UpdateTerminationProtection', {
          onUpdate: {
            action: 'updateTerminationProtection',
            parameters: {
              EnableTerminationProtection: props.terminationProtection,
              StackName: stackName,
            },
            service: 'CloudFormation',
          },
        });
      } else {
        throw Error('Termination protection option must be set to true or false.');
      }
    };
  }
}
