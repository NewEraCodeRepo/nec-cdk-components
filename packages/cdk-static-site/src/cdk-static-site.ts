#!/usr/bin/env node
/* tslint:disable */
import { CloudFrontWebDistribution, SecurityPolicyProtocol, SSLMethod } from '@aws-cdk/aws-cloudfront';
import { HostedZone, ARecord, RecordTarget } from '@aws-cdk/aws-route53';
import { CloudFrontTarget } from '@aws-cdk/aws-route53-targets/lib';
import { Bucket } from '@aws-cdk/aws-s3';
import { BucketDeployment, Source } from '@aws-cdk/aws-s3-deployment';
import { Construct, RemovalPolicy } from '@aws-cdk/core';
import * as path from 'path';

export interface CdkStaticSiteProps {
  bucketConfiguration: {
    bucketName: string;
    removalPolicy: RemovalPolicy;
    websiteIndexDocument: string;
    websiteErrorDocument: string;
    source: string;
  };
  aliasConfiguration: {
    domainName: string;
    siteDomain: string;
    acmCertRef: string;
  };
}

/**
 * Static site infrastructure, using an S3 bucket for the content.
 *
 * The site redirects from HTTP to HTTPS, using a CloudFront distribution,
 * Route53 alias record, and ACM certificate.
 *
 * The ACM certificate is expected to be created and validated outside of the CDK,
 * with the certificate ARN stored in an SSM Parameter.
 */
export class CdkStaticSite extends Construct {
  constructor(parent: Construct, name: string, props: CdkStaticSiteProps) {
    super(parent, name);

    const { bucketName, removalPolicy, websiteIndexDocument, websiteErrorDocument, source } = props.bucketConfiguration;
    const { domainName, siteDomain, acmCertRef } = props.aliasConfiguration;

    // Content bucket
    const siteBucket = new Bucket(this, 'SiteBucket', {
      bucketName,
      publicReadAccess: true,
      // The default removal policy is RETAIN, which means that cdk destroy will not attempt to delete
      // the new bucket, and it will remain in your account until manually deleted. By setting the policy to
      // DESTROY, cdk destroy will attempt to delete the bucket, but will error if the bucket is not empty.
      removalPolicy, // DESTROY NOT recommended for production code
      websiteErrorDocument,
      websiteIndexDocument,
    });

    // CloudFront distribution that provides HTTPS
    const distribution = new CloudFrontWebDistribution(this, 'SiteDistribution', {
      aliasConfiguration: {
        acmCertRef,
        names: [siteDomain],
        securityPolicy: SecurityPolicyProtocol.TLS_V1_1_2016,
        sslMethod: SSLMethod.SNI,
      },
      defaultRootObject: websiteIndexDocument,
      originConfigs: [
        {
          behaviors: [{ isDefaultBehavior: true }],
          s3OriginSource: {
            s3BucketSource: siteBucket,
          },
        },
      ],
    });

    const placeHolderSource = path.join(__dirname, '..', 'website');

    new BucketDeployment(this, 'DeployWebsite', {
      destinationBucket: siteBucket,
      distribution,
      retainOnDelete: removalPolicy === RemovalPolicy.RETAIN,
      sources: [Source.asset(source || placeHolderSource)],
    });

    // Route53 alias record for the CloudFront distribution
    const zone = HostedZone.fromLookup(this, 'Zone', {
      domainName,
    });
    new ARecord(this, 'SiteAliasRecord', {
      recordName: siteDomain,
      target: RecordTarget.fromAlias(new CloudFrontTarget(distribution)),
      zone,
    });
  }
}
