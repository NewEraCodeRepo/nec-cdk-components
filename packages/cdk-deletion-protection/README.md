# NewEraCode Deletion Protection Component

## Description

An [AWS CDK](https://docs.aws.amazon.com/cdk/latest/guide/home.html) construct that protects the stack from being deleted or a resource from being deleted during update or deployment.

## Usage

### Install

```shell
$ npm i @NewEraCode/cdk-deletion-protection
```
### Import

**NodeJS**
```javascript
const cdp = require('@neweracode/cdk-deletion-protection');
const deletionProtection = new cdp.CDKDeletionProtection(....)
```
**TypeScript / Babel**
```typescript
import { CDKDeletionProtection } from '@neweracode/cdk-deletion-protection';
```

Example:

```typescript
import { UserPool, CfnUserPool } from '@aws-cdk/aws-cognito';
import { CDKDeletionProtection } from '@neweracode/cdk-deletion-protection';

const deletionProtection = new CDKDeletionProtection(this, 'ProtectionPolicies', {
    resourceDeletionPolicy: "RETAIN",
    terminationProtection: true,
});

// enable stack termination deletion protection:
deletionProtection.setStackTerminationProtection(this.stackName);

// protect a Cognito User Pool from getting deleted on update:
const userPool = new UserPool(this, 'cdkUserPool', {...});
const cfnUserPool = userPool.node.defaultChild as CfnUserPool;
deletionProtection.setResourcePolicy(cfnUserPool);
```
