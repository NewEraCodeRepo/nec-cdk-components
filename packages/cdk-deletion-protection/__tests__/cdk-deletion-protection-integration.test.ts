import { App, CfnResource, Environment, Stack } from '@aws-cdk/core';
import 'jest-cdk-snapshot';
// tslint:disable-next-line: no-var-requires
import { CDKDeletionProtection, ICDKDeletionProtectionProps } from '../src/cdk-deletion-protection';

function mockDateNow() {
  return 1566592070148;
}

const originalDateNow = Date.now;

describe('cdk-deletion-protection', () => {
  beforeEach(() => {
    Date.now = mockDateNow;
  });

  afterEach(() => {
    Date.now = originalDateNow;
  });

  test('snapshot', () => {
    const env: Environment = {
      account: '123456789012',
      region: 'us-east-1',
    };

    const deletionProtectionProps: ICDKDeletionProtectionProps = {
      resourceDeletionPolicy: 'RETAIN',
      terminationProtection: true,
    };

    const app = new App();
    const stack = new Stack(app, 'Test', { env });
    const deletionProtection = new CDKDeletionProtection(
      stack,
      'InvalidCDKDeletionProtection',
      deletionProtectionProps,
    );

    const cnfResource = new CfnResource(stack, 'PolicyResource', {
      type: 'Test::Resource::ValidPolicy',
    });

    deletionProtection.setResourcePolicy(cnfResource);
    deletionProtection.setStackTerminationProtection(stack.stackName);

    expect(stack).toMatchCdkSnapshot();
  });
});
