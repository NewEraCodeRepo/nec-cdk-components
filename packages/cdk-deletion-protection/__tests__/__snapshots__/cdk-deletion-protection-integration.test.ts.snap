// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cdk-deletion-protection snapshot 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfArtifactHash6EE8789B": Object {
      "Description": "Artifact hash for asset \\"501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cf\\"",
      "Type": "String",
    },
    "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfS3Bucket334043A5": Object {
      "Description": "S3 bucket for asset \\"501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cf\\"",
      "Type": "String",
    },
    "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfS3VersionKey45E8602E": Object {
      "Description": "S3 key for asset version \\"501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cf\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "AWS679f53fac002430cb0da5b7982bd22872D164C4C": Object {
      "DependsOn": Array [
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleDefaultPolicyD28E1A5E",
        "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfS3Bucket334043A5",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfS3VersionKey45E8602E",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters501b56884b5970eb66e5eb6ba2f489d9456188836fd49044647e94a28a0ac2cfS3VersionKey45E8602E",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 120,
      },
      "Type": "AWS::Lambda::Function",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleDefaultPolicyD28E1A5E": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "cloudformation:UpdateTerminationProtection",
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleDefaultPolicyD28E1A5E",
        "Roles": Array [
          Object {
            "Ref": "AWS679f53fac002430cb0da5b7982bd2287ServiceRoleC1EA0FF2",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InvalidCDKDeletionProtectionUpdateTerminationProtectionA84119A2": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "Create": Object {
          "action": "updateTerminationProtection",
          "parameters": Object {
            "EnableTerminationProtection": "TRUE:BOOLEAN",
            "StackName": "Test",
          },
          "physicalResourceId": "1566592070148",
          "service": "CloudFormation",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "AWS679f53fac002430cb0da5b7982bd22872D164C4C",
            "Arn",
          ],
        },
        "Update": Object {
          "action": "updateTerminationProtection",
          "parameters": Object {
            "EnableTerminationProtection": "TRUE:BOOLEAN",
            "StackName": "Test",
          },
          "physicalResourceId": "1566592070148",
          "service": "CloudFormation",
        },
      },
      "Type": "Custom::AWS",
      "UpdateReplacePolicy": "Delete",
    },
    "PolicyResource": Object {
      "DeletionPolicy": "Retain",
      "Type": "Test::Resource::ValidPolicy",
      "UpdateReplacePolicy": "Retain",
    },
  },
}
`;
